qiskit/backends/local/statevector_simulator_py.py

```diff
@@ -87,15 +89,16 @@ def _validate(self, qobj):
         1. No shots
         2. No measurements in the middle
         """
-        if qobj['config']['shots'] != 1:
+        if qobj.config.shots != 1:
             logger.info("statevector simulator only supports 1 shot. "
                         "Setting shots=1.")
-            qobj['config']['shots'] = 1
-        for circuit in qobj['circuits']:
+            qobj.config.shots = 1
+        for circuit_qobj in qobj.circuits:
+            circuit = circuit_qobj.as_dict()
             if 'shots' in circuit['config'] and circuit['config']['shots'] != 1:
                 logger.info("statevector simulator only supports 1 shot. "
                             "Setting shots=1 for circuit %s.", circuit['name'])
-                circuit['config']['shots'] = 1
+                circuit_qobj.config.shots = 1
             for op in circuit['compiled_circuit']['operations']:
                 if op['name'] in ['measure', 'reset']:
```

Comment:
Shouldn't these lines be updated also to work with object not dictionary?

Reply:
Yep, fixed!